macros/parsers/parserMultiAnswer.pl

```diff
@@ -89,15 +91,33 @@ sub setCmpFlags {
 #  the individual answer checkers.
 #
 sub cmp {
-	my $self    = shift;
-	my %options = @_;
+	my ($self, %options) = @_;
+
+	%options = (%options, %{ $self->{cmpOpts} }) if ref($self->{cmpOpts}) eq 'HASH';
+
 	foreach my $id ('checker', 'separator') {
 		if (defined($options{$id})) {
 			$self->{$id} = $options{$id};
 			delete $options{$id};
 		}
 	}
-	die "You must supply a checker subroutine" unless ref($self->{checker}) eq 'CODE';
+
+	unless (ref($self->{checker}) eq 'CODE') {
```

Comment:
You may wish to produce an error message if `checker` is defined but not a code reference, rather than just replacing it silently.

Reply:
Thanks for the suggestion.

Reply:
I think you could put the check inside the `unless` clause, as 

``` perl
unless (ref($self->{checker} eq 'CODE') {
  die "Your checker must be a subroutine." if defined($self->{checker});
```

instead of having to check for `CODE` two times.